x/migration/types/morse_validation.go

```diff
@@ -0,0 +1,70 @@
+package types
+
+import (
+	"crypto/ed25519"
+
+	cmted25519 "github.com/cometbft/cometbft/crypto/ed25519"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+)
+
+// morseClaimMessage is an interface that all Morse account/actor claim messages
+// implement which allows for a non-generic implementations of common behavior.
+type morseClaimMessage interface {
```

Comment:
A few notes about the design of this:
1. The functions here don't use the keeper
2. The functions do "basic validation"
3. We are introducing an interface for a proto msg

Why don't we follow the same structure we have elsewhere like `func (msg *MsgStakeApplication) ValidateBasic() error {`?

Reply:
> 1. The functions here don't use the keeper

Correct! Since the public key is now part of the Morse account/actor claim messages (and no longer persisted onchain)

> 2. The functions do "basic validation"

I would define this as "all validation which doesn't require onchain data"; which now applies to Morse signature verification.

> 3. We are introducing an interface for a proto msg

Indeed. Since there are multiple Morse account/actor claim messages, and this is such critical logic, consolidating it requires the use of an interface.

Reply:
Update.

I saw how this is used.

Do you see an opportunity to:
1. Enforce a certain type that implements this interface so it's explicit where it's used?
2. #PUC on which messages its used for.